issue_1/contracts/WordRegistry.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.4;
+
+import "hardhat/console.sol";
+
+import "./yield-utils-v2/contracts/access/Ownable.sol";
+
+/**
+@title Dictatorship-friendly word registry
+@author Owleksiy
+@notice This contract allows users to register/unregister (claim/unclaim) words
+Once a word is regiistered by a user, other users can't register it, unless the
+original user unregisters it.
+
+Additionally, the owner of the contract can expropriate any registered word
+ */
+contract WordRegistry is Ownable {
+    event OwnershipChange(string word, address previousOwner, address newOwner);
+    // word -> owner mapping
+    mapping(string => address) private word_to_owner;
+
+    constructor() {
+        console.log("Creating a registry, owner:", msg.sender);
+    }
+
+    function _is_owned_by_somebody(string memory word) private view returns (bool) {
+        return word_to_owner[word] != address(0);
+    }
+
+    function _transfer(
+        string memory word,
+        address old_owner,
+        address new_owner
+    ) private {
+        word_to_owner[word] = new_owner;
+        emit OwnershipChange(word, old_owner, new_owner);
+        console.log("Registered: '", word, "' to: ", new_owner);
+    }
+
+    /**
+    @notice Checks if msg.sender owns the word
+    @param word word to check
+    @return ownership status
+     */
+    function does_own(string memory word) public view returns (bool) {
```

Comment:
It would be simpler to expose the mapping to the outside world. All public contract variables are accessible as if they would be `view` functions:

`mapping(string => address) public word_to_owner;`

Can be accessed as:

`function word_to_owner(string) public view returns (address);`

Reply:
80% ack (hiding internal state behind interfaces feels like the right thing to do - if I want to rename my variable, the users don't have to worry about it; the compiler should optimize the getter out, I guess?). Will remove this method, but I'll do a bit of digging to confirm if explicit getters are a bad idea.

Reply:
The thing is that you won't rename the variable, ever. Once the contract is deployed it's there forever. Smart contract development relies heavily on minimalism both for economic reasons (everything costs gas, and users are very sensitive to it) and for security reasons (complexity hides bugs, that you can't fix easily once deployed)

Reply:
ah, the contract being immutable is a great point